boxsdk/session/box_session.py

```diff
@@ -336,68 +312,57 @@ def _prepare_and_send_request(
             `int`
         """
         files = kwargs.get('files')
-        file_stream_positions = None
+        kwargs['file_stream_positions'] = None
         if files:
-            file_stream_positions = dict((name, file_tuple[1].tell()) for name, file_tuple in files.items())
-        return self._make_request(
-            method,
-            url,
-            headers,
-            auto_session_renewal,
-            expect_json_response,
-            attempt_number,
-            file_stream_positions=file_stream_positions,
-            **kwargs
+            kwargs['file_stream_positions'] = dict((name, file_tuple[1].tell()) for name, file_tuple in files.items())
+        attempt_number = 0
+
+        request = self.box_request_constructor(
+            url=url,
+            method=method,
+            headers=headers,
+            auto_session_renewal=auto_session_renewal,
+            expect_json_response=expect_json_response,
         )
 
-    def _make_request(
-            self,
-            method,
-            url,
-            headers=None,
-            auto_session_renewal=True,
-            expect_json_response=True,
-            attempt_number=0,
-            **kwargs
-    ):
+        network_response = self._send_request(request, **kwargs)
+
+        while True:
+            retry = self._get_retry_request_callable(network_response, attempt_number, request)
+
+            if retry is None or attempt_number >= 10:
```

Comment:
The commit message was a bit confusing:
```
we'll give up after 10 failures regardless
of whether they are 5xx responses or 429s
```
because 202's and 401's are in play. The code here gives on up an y `retriable`failure after 10 attempts